src/diffusers/pipelines/animatediff/pipeline_animatediff.py

```diff
@@ -695,52 +727,111 @@ def __call__(
         extra_step_kwargs = self.prepare_extra_step_kwargs(generator, eta)
-        # 7 Add image embeds for IP-Adapter
+
+        # 7. Add image embeds for IP-Adapter
         added_cond_kwargs = {"image_embeds": image_embeds} if ip_adapter_image is not None else None
 
-        # Denoising loop
-        num_warmup_steps = len(timesteps) - num_inference_steps * self.scheduler.order
-        with self.progress_bar(total=num_inference_steps) as progress_bar:
-            for i, t in enumerate(timesteps):
-                # expand the latents if we are doing classifier free guidance
-                latent_model_input = torch.cat([latents] * 2) if do_classifier_free_guidance else latents
-                latent_model_input = self.scheduler.scale_model_input(latent_model_input, t)
-
-                # predict the noise residual
-                noise_pred = self.unet(
-                    latent_model_input,
-                    t,
-                    encoder_hidden_states=prompt_embeds,
-                    cross_attention_kwargs=cross_attention_kwargs,
-                    added_cond_kwargs=added_cond_kwargs,
-                ).sample
-
-                # perform guidance
-                if do_classifier_free_guidance:
-                    noise_pred_uncond, noise_pred_text = noise_pred.chunk(2)
-                    noise_pred = noise_pred_uncond + guidance_scale * (noise_pred_text - noise_pred_uncond)
-
-                # compute the previous noisy sample x_t -> x_t-1
-                latents = self.scheduler.step(noise_pred, t, latents, **extra_step_kwargs).prev_sample
-
-                # call the callback, if provided
-                if i == len(timesteps) - 1 or ((i + 1) > num_warmup_steps and (i + 1) % self.scheduler.order == 0):
-                    progress_bar.update()
-                    if callback is not None and i % callback_steps == 0:
-                        callback(i, t, latents)
+        # 7.1 FreeInit
+        freeinit_num_steps = freeinit_kwargs.get("num_steps", 5) if use_freeinit else 1
+        freeinit_use_fast_sampling = freeinit_kwargs.get("use_fast_sampling", False)
+
+        if use_freeinit:
+            if "method" not in freeinit_kwargs.keys():
+                raise ValueError("`use_freeinit` was set to True but required freeinit kwarg `method` was not set")
+
+            freeinit_filter_shape = (
+                1,
+                num_channels_latents,
+                num_frames,
+                height // self.vae_scale_factor,
+                width // self.vae_scale_factor,
+            )
+            freeinit_freq_filter = get_freeinit_freq_filter(
+                shape=freeinit_filter_shape,
+                device=device,
+                filter_type=freeinit_kwargs.get("method"),
+                n=freeinit_kwargs.get("n", 4) if freeinit_kwargs.get("method") == "butterworth" else None,
+                d_s=freeinit_kwargs.get("d_s", 0.25),
```

Comment:
Maybe we could rename `d_s` and `d_t` to something like `spatial_stop_frequency` and `temporal_stop_frequency` or better?